### PR TITLE
HERA-250 [Questions] upgrades questions grading and rendering functionality

### PR DESCRIPTION
**Pull request by Golub-Sergey**
**Description:** upgrades questions grading and rendering functionality

**Youtrack:** [ticket](https://youtrack.raccoongang.com/issue/HERA-250), [review ticket](https://youtrack.raccoongang.com/issue/HERA-253)

### CODE REVIEW

File: hera_xblocks/question/question/static/html/question.html [hunk 1]
```diff
@@ -2,67 +2,95 @@
...
+                % if break_down.get('imgUrls'):
+                    %for url in break_down['imgUrls']:
+                        <% break_down_url += url + ' ' %>
+                    % endfor
+                %endif
+                % if teach_me.get('imgUrls'):
+                    %for url in teach_me['imgUrls']:
+                        <% teach_me_url += url + ' ' %>
+                    % endfor
+                %endif
```
Comment (sendr) [inline hera_xblocks/question/question/static/html/question.html]:
```suggestion
                <% teach_me_url = ' '.join(teach_me.get('imgUrls', [])) %>
```

Comment (sendr) [reply hera_xblocks/question/question/static/html/question.html]: and others the same way please

File: hera_xblocks/question/question/question.py [hunk 2]
```diff
@@ -136,37 +126,40 @@ def get_data(self, somedata, sufix=''):
...
+        for index, question in enumerate(self.problem_types):
 
-        self.user_answer = data.get("answer")
+            if question['type'] == "number":
+                answer = False
+                if answers[index][0]:
+                    temp_answer = float(answers[index][0])
+                    correct_answer = float(question['answer'])
+                    preciseness = self.preciseness(question)
+                    answer = correct_answer - preciseness <= temp_answer <= correct_answer + preciseness
```
[the remark this answers is not in the document]
Comment (Golub-Sergey) [reply hera_xblocks/question/question/question.py]: Done

File: hera_xblocks/question/question/question.py [hunk 3]
```diff
@@ -136,37 +126,40 @@ def get_data(self, somedata, sufix=''):
...
+                correct_answers.append(answer)
+
+            elif question['type'] == "text":
+                answer = ''.join(answers[index]) == question['answer']
```
[the remark this answers is not in the document]
Comment (Golub-Sergey) [reply hera_xblocks/question/question/question.py]: Done

File: hera_xblocks/question/question/question.py [hunk 4]
```diff
@@ -136,37 +126,40 @@ def get_data(self, somedata, sufix=''):
...
+                correct_answers.append(answer)
+
+            elif question['type'] in ["select", "radio", "checkbox"]:
+                temp_answers = [option["title"] for option in question['options'] if option["correct"] is True]
```
[the remark this answers is not in the document]
Comment (Golub-Sergey) [reply hera_xblocks/question/question/question.py]: Done

File: hera_xblocks/question/question/static/html/question.html [hunk 5]
```diff
@@ -2,67 +2,95 @@
...
                     </span>
                     Rephrase
                 </button>
-                <button type="button" class="scaffold__btn is-break" data-img-urls="{%for url in break_down.imgUrls%}{{url}} {% endfor %}" data-content="{{break_down.content}}">
+                <%
+                    break_down_url = ''
+                    teach_me_url = ''
+                %>
+                % if break_down.get('imgUrls'):
+                    %for url in break_down['imgUrls']:
```
[the remark this answers is not in the document]
Comment (Golub-Sergey) [reply hera_xblocks/question/question/static/html/question.html]: Done

File: hera_xblocks/question/question/static/html/question.html (hunk 1 above)
Comment (Golub-Sergey) [reply hera_xblocks/question/question/static/html/question.html]: Done

File: hera_xblocks/question/question/question.py [hunk 6]
```diff
@@ -40,28 +40,19 @@ def description(self):
         return self.data.get("description")
 
     @property
-    def question(self):
-        return self.data.get("question")
+    def problem_types(self):
+        return self.data.get("problemTypes")
```
Comment (sendr) [inline hera_xblocks/question/question/question.py]:
```suggestion
        return self.data.get("problemTypes", [])
```

File: hera_xblocks/question/question/static/html/question.html [hunk 7]
```diff
@@ -2,67 +2,85 @@
...
-                <button type="button" class="scaffold__btn is-rephrase" data-content="{{rephrase.content}}">
+                <button type="button" class="scaffold scaffold__btn is-rephrase" data-content="${rephrase['content']}">
                     <span class="scaffold__price">
                         -20
                     </span>
                     Rephrase
                 </button>
-                <button type="button" class="scaffold__btn is-break" data-img-urls="{%for url in break_down.imgUrls%}{{url}} {% endfor %}" data-content="{{break_down.content}}">
+                <%
+                    break_down_img_url = ' '.join(break_down.get('imgUrls', []))
```
Comment (sendr) [inline hera_xblocks/question/question/static/html/question.html]:
```suggestion
                    break_down_img_urls = ' '.join(break_down.get('imgUrls', []))
```

File: hera_xblocks/question/question/static/html/question.html [hunk 8]
```diff
@@ -2,67 +2,85 @@
...
+                    teach_me_img_url = ' '.join(teach_me.get('imgUrls', []))
```
Comment (sendr) [inline hera_xblocks/question/question/static/html/question.html]:
```suggestion
                    teach_me_img_urls = ' '.join(teach_me.get('imgUrls', []))
```

File: hera_xblocks/question/question/question.py (hunk 6 above)
Comment (Golub-Sergey) [reply hera_xblocks/question/question/question.py]: Done

File: hera_xblocks/question/question/static/html/question.html (hunk 7 above)
Comment (Golub-Sergey) [reply hera_xblocks/question/question/static/html/question.html]: Done

File: hera_xblocks/question/question/static/html/question.html (hunk 8 above)
Comment (Golub-Sergey) [reply hera_xblocks/question/question/static/html/question.html]: Done

**Pull request merged**